src/Strategy/Go/GoModGraph.hs

```diff
@@ -179,16 +192,36 @@ analyze dir = do
   goModGraphStdout <- context ("Getting selected dependencies versions using, " <> toText (show goModGraphCmd)) $ execParser parseGoModGraph dir goModGraphCmd
   ggm <- fromMaybeText "expected to find main module, but found no main module!" mainMod
 
-  let graph = buildGraph goModGraphStdout ggm selectedMods directMods filterModsNotUsedInBuild
+  let graph = buildGraph goModGraphStdout ggm selectedMods directMods filterModsNotUsedInBuild modReplacements
   pure (graph, Complete)
   where
+    -- TODO: need to convert the version from the replacement into something well-typed
```

Comment:
If this does not require significantly more cycles - I rather we do this as part of this PR.

Reply:
This is actually a holdover comment I forgot to add. I do this step in `toGraphModWithReplacement`. I will remove the comment - is there any type safety you think is missing that should be added in addition?